db/collab.go

```diff
@@ -17,13 +17,17 @@ import (
 type Message struct {
 	Author string `json:"author"`
 	Type   string `json:"type"`
+	Target string `json:"target"`
 	Body   string `json:"body"`
 }
+type StringSet map[string]bool
 type Session struct {
 	// Map UIDs to their websocket.Conn
 	Conns   map[string]*websocket.Conn
 	Teacher string
```

Comment:
What do you think about extending the websocket `Conn` type with our own fields to simplify the code a bit? It would allow set broadcast functions to be localized to per-conection bases. Here's a suggested starter for the signature:

```
type Connection struct {
    *websocket.Conn
    UID string
    Subscriptions []string
}

type Session struct {
	// Map UIDs to their websocket.Conn
	Conns   map[string]Connection
```

I'm not too particular either way, just wanted to field some thoughts.